src/test/kotlin/org/rust/ide/navigation/goto/RsGotoSuperHandlerTest.kt

```diff
@@ -72,17 +73,45 @@ class RsGotoSuperHandlerTest : RsTestBase() {
         }
     """)
 
-    fun testOnFileLevel() {
-        fileTreeFromText("""
-            //- foo.rs
-            /*caret*/    // only comment
+    fun `test on file level`() = checkNavigationInFiles("""
+        //- foo.rs
+        /*caret*/    // only comment
 
-            //- main.rs
-                mod foo;
-        """).createAndOpenFileWithCaretMarker()
+        //- main.rs
+            mod foo;
+        """, expected = "mod foo;")
 
+    fun `test with path attribute`() = checkNavigationInFiles("""
+        //- foo.rs
+        /*caret*/    // only comment
+
+        //- main.rs
+            #[path="foo.rs"]
+            mod bar;
+
+    """, expected = """
+        #[path="foo.rs"]
+        mod bar;
+    """)
+
+    fun `test with path attribute in module`() = checkNavigationInFiles("""
+        //- foo/qqq.rs
+        /*caret*/    // only comment
+
+        //- main.rs
+        mod foo {
+            #[path="qqq.rs"]
+            mod bar;
+        }
+    """, expected = """
+        #[path="qqq.rs"]
+            mod bar;
```

Comment:
Indentation is off here.

Reply:
Yeah, but target text has this weird indent too. I don't really know why

Reply:
Ah,  because we are looking at `element.text`, which includes indentation for all but the first line!

Yeah, let's leave this as is, though it would be cool to invent some better way to mark target elements for tests.

Reply:
It would be cool to compare file tree before and after as we do in single file case

Reply:
Hm, yeah, I think we can use file tree to compare caret positions!